src/pages/dataElements/edit/createJsonPatchOperations.ts

```diff
@@ -1,50 +1,69 @@
 import get from 'lodash/fp/get'
 import { JsonPatchOperation } from '../../../types'
-import { DataElement } from '../../../types/generated'
-import type { FormValues } from '../form'
+import { Attribute, AttributeValue } from './../../../types/generated/models'
 
-type DataElementKey = keyof DataElement
+type PatchAttributeFields = {
+    id: Attribute['id']
+}
+
+type PatchAttributeValue = {
+    attribute: PatchAttributeFields
+    value: AttributeValue['value']
+}
 
-interface FormatFormValuesArgs {
-    dataElement: DataElement
-    dirtyFields: { [name: string]: boolean }
+type ModelWithAttributeValues = {
+    attributeValues: PatchAttributeValue[]
+}
+
+interface FormatFormValuesArgs<FormValues extends ModelWithAttributeValues> {
+    originalValue: unknown
```

Comment:
This could potentially be a bit safer. Currently since we just check with `get` the type here doesn't matter too much. However I'm not sure if it would be correct that eg. this should "extend" `FormValues` (or the other way around). It could potentially also extend `Partial<BaseIdentifiableObject>`.